ui/components/ui/box/box.d.ts

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+
+import {
+  AlignItems,
+  BLOCK_SIZES,
+  BorderStyle,
+  BackgroundColor,
+  BorderColor,
+  TextColor,
+  IconColor,
+  DISPLAY,
+  JustifyContent,
+  TEXT_ALIGN,
+  FLEX_DIRECTION,
+  FLEX_WRAP,
+  BorderRadius,
+} from '../../../helpers/constants/design-system';
+
+export type BoxChildren = React.ReactNode | ((...args: any[]) => any);
+
+export type BoxFlexDirection = typeof FLEX_DIRECTION;
+export type BoxFlexDirectionArray = [
+  BoxFlexDirection[keyof BoxFlexDirection],
+  BoxFlexDirection[keyof BoxFlexDirection]?,
+  BoxFlexDirection[keyof BoxFlexDirection]?,
+  BoxFlexDirection[keyof BoxFlexDirection]?,
+];
```

Comment:
Can you walk me through a bit of this array here, it seems like an array with repeated elements?

Reply:
Yes! So all Box props can be passed a single value OR an array of values which act as responsive props

```jsx
<Box display={DISPLAY.FLEX} />

// or

<Box display={[DISPLAY.NONE, DISPLAY.FLEX]} />
```
`display: none` 0px - 575px
`display: flex` 576px - and up

[Docs here](https://metamask.github.io/metamask-storybook/?path=/docs/components-ui-box--responsive-props#responsive-props) not sure why the link doesn't go to this section. Scroll down until you see "Responsive props" 
<img width="1440" alt="Screenshot 2023-03-06 at 4 57 17 PM" src="https://user-images.githubusercontent.com/8112138/223291423-d3f3f05f-8332-4a54-960d-52b48731cb6a.png">

The type should allow for a single value or an array with a minimum of one item and a maximum of 4 items

Reply:
Super clear, thank you !